### PR TITLE
ISPN-6542 Subclasses of AbstractWriteKeyCommand don't marshal the Params params field

### PR DESCRIPTION
**Pull request by rvansa**
* cache enum values
* replaced externalizer with static methods

### CODE REVIEW

File: commons/src/main/java/org/infinispan/commons/api/functional/Param.java [hunk 1]
```diff
@@ -128,6 +129,10 @@ public PersistenceMode get() {
       public static PersistenceMode defaultValue() {
          return PERSIST;
       }
+
+      public static PersistenceMode fromOrdinal(int ordinal) {
```
Comment (danberindei) [inline commons/src/main/java/org/infinispan/commons/api/functional/Param.java]: The other enums have `valueOf(int)`.

**Comment by danberindei** [comment]
Looks good, the only change I'd suggest is the `fromOrdinal` method name.

**Comment by rvansa** [comment]
@danberindei ok, I've noticed just the `ordinal -> Xxx.CACHED_VALUES[ordinal]` lambdas. And `fromOrdinal` is a better name, but I'll stay consistent. Updated.

**Comment by danberindei** [comment]
Integrated, thanks Radim!

**Pull request closed without merging**